torch/ao/nn/quantized/modules/linear.py

```diff
@@ -309,10 +309,14 @@ def from_float(cls, mod, use_precomputed_fake_quant=False):
             # the type mismatch in assignment. Also, mypy has an issue with
             # iterables not being implemented, so we are ignoring those too.
             if not isinstance(cls._FLOAT_MODULE, Iterable):
-                cls._FLOAT_MODULE = [cls._FLOAT_MODULE]  # type: ignore[assignment]
-            supported_modules = ", ".join([float_mod.__name__ for float_mod in cls._FLOAT_MODULE])  # type: ignore[attr-defined]
+                cls._FLOAT_MODULE = [cls._FLOAT_MODULE]
+            supported_modules = ", ".join(
+                [float_mod.__name__ for float_mod in cls._FLOAT_MODULE]
+            )
             error_msg = f"nnq.{cls.__name__}.from_float only works for {supported_modules}, but got: {type(mod)}"
-            assert type_before_parametrizations(mod) in cls._FLOAT_MODULE, error_msg.format()  # type: ignore[attr-defined]
+            assert (
+                type_before_parametrizations(mod) in cls._FLOAT_MODULE
+            ), error_msg.format()
```

Comment:
Not, should error_msg be wrapped in our LazyStr so the formatting only happens when the assert triggers?

Reply:
That sounds sensible, but I may be lacking context to judge this.

In any case this change should probably not be made as part of this PR right? This PR is just about remove type ignores that aren't needed, trying not to mix it with any behavioral changes.